nomadproxy.go

```diff
@@ -132,12 +155,51 @@ func main() {
 	}
 
 	log.Printf("nomadproxy running at %v, proxying to %v", ln.Addr(), *backendAddr)
-	log.Fatal(http.Serve(ln, auditRequests(localClient, proxy)))
+	log.Fatal(http.Serve(ln, auditRequests(ctx, localClient, proxy)))
+}
+
+func newOauthClient() *tailscale.Client {
+	tailnet := os.Getenv("TAILNET_NAME")
+	clientID := os.Getenv("TS_OAUTH_CLIENT_ID")
+	clientSecret := os.Getenv("TS_OAUTH_CLIENT_SECRET")
+
+	if tailnet == "" || clientID == "" || clientSecret == "" {
+		log.Fatal("missing OAuth client configuration (TAILNET_NAME, TS_OAUTH_CLIENT_ID, TS_OAUTH_CLIENT_SECRET)")
+	}
+
+	return &tailscale.Client{
+		Tailnet: tailnet,
+		HTTP: tailscale.OAuthConfig{
+			ClientID:     clientID,
+			ClientSecret: clientSecret,
+			Scopes:       []string{"auth_keys:write"},
+		}.HTTPClient(),
+	}
+}
+
+func createAuthKey(ctx context.Context, client *tailscale.Client, tags []string) (string, error) {
+	capabilities := tailscale.KeyCapabilities{}
+	capabilities.Devices.Create.Ephemeral = true
+	capabilities.Devices.Create.Reusable = false
+	capabilities.Devices.Create.Preauthorized = false
```

Comment:
This... might want to be configurable - because otherwise some types of tailnet locking might require manual device approval, but... it's fine for now - we can always change it in a follow up